src/credentials/GroupDataProvider.h

```diff
@@ -262,6 +284,28 @@ class GroupDataProvider
 
     // General
     virtual CHIP_ERROR Decrypt(PacketHeader packetHeader, PayloadHeader & payloadHeader, System::PacketBufferHandle & msg) = 0;
+
+    // Listener
+    void SetListener(Listener * listener) { mListener = listener; };
+    void RemoveListener() { mListener = nullptr; };
+
+protected:
+    void GroupAdded(chip::FabricIndex fabric_index, const GroupInfo & new_group)
+    {
+        if (mListener)
+        {
+            mListener->OnGroupAdded(fabric_index, new_group);
+        }
+    }
+    void GroupRemoved(chip::FabricIndex fabric_index, const GroupInfo & old_group)
```

Comment:
This is called in only one place. maybe we could inline it there directly.

Reply:
Inline now.